toolchain/lex/lex.cpp

```diff
@@ -858,6 +860,75 @@ auto Lexer::LexCommentOrSlash(llvm::StringRef source_text, ssize_t& position)
   CARBON_CHECK(result, "Failed to form a token!");
 }
 
+// Lexes for a comment prefix using SIMD support. Returns true if the end of
+// lines with the provided prefix has been found. Returns false if it's unable
+// to finish lexing (but may be partially lexed).
+//
+// TODO: We should extend this to 32-byte SIMD on platforms with support.
+static auto LexCommentPrefixInSimd(llvm::StringRef source_text,
+                                   ssize_t& position, ssize_t prefix_size,
+                                   ssize_t first_line_start,
+                                   llvm::function_ref<void()> skip_to_next_line)
```

Comment:
Could you split out the refactoring into a separate commit or PR?

Because this PR *does* contain a behavior change, it makes it harder for me to review this section. The code here is fairly complex, and there seem to be three things I need to look for:

1) Extracting it to a separate function
2) Changing how the different architectures are grouped in the code
3) (possibly) some behavior change to support the stitching below.

And it's hard to review all three without the incremental diff for (2) or at least (3).

FWIW, I find the original structure of the code easier to read -- the duplication of the loop condition and call to `skip_to_next_line`, for me, are outweighed by being able to see the complete loop for each architecture.

Reply:
Reverted.